src/metabase/lib/field.cljc

```diff
@@ -359,19 +364,22 @@
   (lib.binning/available-binning-strategies query stage-number (resolve-field-metadata query stage-number field-ref)))
 
 (defmethod lib.binning/available-binning-strategies-method :metadata/field
-  [query _stage-number {:keys [effective-type fingerprint semantic-type] :as _field-metadata}]
-  (let [binning?        (some-> query lib.metadata/database :features (contains? :binning))
-        {min-value :min max-value :max} (get-in fingerprint [:type :type/Number])]
-    (cond
-      ;; TODO: Include the time and date binning strategies too; see metabase.api.table/assoc-field-dimension-options.
-      (and binning? min-value max-value
-           (isa? semantic-type :type/Coordinate))
-      (lib.binning/coordinate-binning-strategies)
-
-      (and binning? min-value max-value
-           (isa? effective-type :type/Number)
-           (not (isa? semantic-type :Relation/*)))
-      (lib.binning/numeric-binning-strategies))))
+  [query _stage-number {:keys [effective-type fingerprint semantic-type] :as field-metadata}]
+  (let [binning?    (some-> query lib.metadata/database :features (contains? :binning))
+        fingerprint (get-in fingerprint [:type :type/Number])
+        existing    (lib.binning/binning field-metadata)
+        strategies  (cond
+                      ;; Abort if the database doesn't support binning, or this column does not have a defined range.
+                      (not (and binning?
+                                (:min fingerprint)
+                                (:max fingerprint)))               nil
+                      (isa? semantic-type :type/Coordinate)        (lib.binning/coordinate-binning-strategies)
+                      (and (isa? effective-type :type/Number)
+                           (not (isa? semantic-type :Relation/*))) (lib.binning/numeric-binning-strategies))]
```

Comment:
This looks like the type of logic the original version of `metabase.lib.types.isa` was created for. I'm a bit afraid we'll sprinkle such things all over the place and it will be difficult to track them all down when we want to change something.